docs/source/index.rst

```diff
@@ -48,7 +48,17 @@ API References
    compliance.kaldi
    kaldi_io
    utils
+
+Prototype API References
```

Comment:
curious if there's some reason we're making prototypes it's own larger section as opposed to just regrouping within the existing `torchaudio.prototypes` section similar to how https://pytorch.org/audio/main/backend.html is structured?

Reply:
It's somewhat technical and somewhat subjective.

Sphinx-gallery is not flexible enough to nicely render structure with more than two depth. I tried using `.. include:` directive to put all the contents in one page, then the `Reference` did not work, so I ended up using `.. toctree:` directive which created the front page for `torchaudio.prototype`, instead of embedding the contents in one single page.

Then, I stopped and gave some thoughts. 

The biggest difference between modules in `prototype` and others is that each module in `prototype` can contain APIs on different level of abstraction by grouped by topics, rather than functionalities. For example, `rnnt` now has APIs for models and pipelines. (`torchaudio.backend` module is somewhat elaborated, but after all, all the components listed there compose the I/O functionality.) So if we put everything in one page under `torchaudio.prototypes` then that page has categories for topics and features beneath them, which does not have a good, universal way to organize. So when I looked at the new index for the prototype features, I thought that would make sense as well.

Now regarding the new `Prototype API References` index on the left bar, I was originally listing them after the library modules, but having multiple of `torchaudio.prototype.XXX` is cluttering the whole index.